src/v/lsm/core/internal/options.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "base/format_to.h"
+#include "base/units.h"
+#include "lsm/core/compression.h"
+#include "lsm/core/internal/files.h"
+
+#include <cstddef>
+
+namespace lsm::internal {
+
+struct options {
+    struct level_config {
+        // The level number in the database.
+        internal::level number;
+
+        fmt::iterator format_to(fmt::iterator) const;
+    };
+    constexpr static auto default_max_level = 6_level;
+    static constexpr std::vector<level_config> make_default_levels() {
+        std::vector<level_config> levels;
+        levels.reserve(default_max_level() + 1);
+        for (auto lvl = 0_level; lvl <= default_max_level; ++lvl) {
+            levels.emplace_back(lvl);
+        }
+        return levels;
+    }
+    // The levels and their configuration in the database,
+    // this will be sorted by level number and also will be monotonically
+    // increasing from level 0 to level N (configurable).
+    std::vector<level_config> levels = make_default_levels();
+    internal::level max_level() const { return levels.back().number; }
+
+    // At what point do we start throttling writes?
+    constexpr static size_t default_level_zero_slowdown_writes_trigger = 8;
+    size_t level_zero_slowdown_writes_trigger
+      = default_level_zero_slowdown_writes_trigger;
+    // At what point do we halt writes?
+    constexpr static size_t default_level_zero_stop_writes_trigger = 12;
```

Comment:
nit: some units or additional detail about the units would be helpful here (and the other triggers below)

Reply:
done